Sources/Controllers/SpotifyController.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  SpotifyController.swift
+//  ControlKit
+//
+
+import OSLog
+import SpotifyiOS
+import SwiftUI
+
+/// Wrapper for ``SPTAppRemote``  - use this to control playback and subscribe to the playback state of the Spotify app.
+///
+/// `SpotifyController` facilitates the connection to and interaction with the Spotify app, enabling playback control and handling the authorization flow.
+///
+/// - Parameters:
+///   - config: The `SpotifyConfig` object used to configure the controller.
+///   If `config` is `.empty`, the controller will not attempt to connect automatically.
+///   The default log level is ``SPTAppRemoteLogLevel//debug``.
+///   - accessTokenDAO: An object conforming to `DAO<String>` that manages the storage and retrieval of the Spotify access token. By default, this uses a `UserDefaultsDAO` with a key of `"SpotifyAccessToken"`.
+///   - autoConnect: A Boolean value that determines whether the controller should attempt to connect to Spotify automatically upon initialization.
+///   The default value is `true`. If `true` and `config` is not `.empty`, the controller will initiate the connection.
+///
+/// - Warning: Saving the Spotify access token using **``UserDefaults``** is not recommended for production apps 🚨
+///   Prefer a `DAO` that persists values to the Keychain 🔐
+///
+/// - Important: Ensure that the provided `SpotifyConfig` contains valid client ID and redirect URL parameters for proper functionality.
+///
+public final class SpotifyController: NSObject, ObservableObject {
+    
+    @Published public var isPlaying: Bool = false
+    
+    private lazy var remote: SPTAppRemote = {
+        let remote = SPTAppRemote(
+            configuration: SPTConfiguration(
+                clientID: config.clientID,
+                redirectURL: config.redirectURL
+            ),
+            logLevel: config.logLevel
+        )
+        remote.connectionParameters.accessToken = try? accessTokenDAO.get() // Non-blocking failure
+        remote.delegate = self
+        remote.playerAPI?.delegate = self
+        return remote
+    }()
+    
+    private let config: SpotifyConfig
+    private let accessTokenDAO: any DAO<String>
+    
+    public init(
+        config: SpotifyConfig,
+        accessTokenDAO: any DAO<String> = UserDefaultsDAO("SpotifyAccessToken"),
```

Comment:
It may be preferable to remove the `UserDefaultsDAO(“SpotifyAccessToken”)` argument from the default arguments, as the user may not be aware of this section of the description. While the `accessTokenDAO` parameter is required, it raises a question and necessitates reading the documentation. 🐼

```suggestion
        accessTokenDAO: any DAO<String>,
```

Reply:
Thanks for calling this out! It certainly raised a lot more questions for me about how the API should be designed...

In the end, I think I'd like to make the `accessTokenDAO` property optional and have the default parameter in the initialiser be `nil`. This way the access token won't be persisted across app launches and the user has to explicitly provide their own DAO. I've updated the readme and doc comments to reflect this change.

This might be the best compromise between ease-of-use and security, wdyt? 👷